src/core/router/transports/session.h

```diff
@@ -70,6 +70,18 @@ class SignedData {
     m_Stream.write(reinterpret_cast<char *>(&type), sizeof(T));
   }
 
+  /// @brief Insert at a specific position
+  /// @param way beginning, current or end
+  /// @param offset value relative to the seekdir parameter
+  /// @param buf buffer
+  /// @param len length
+  template <typename T>
+  void InsertAt(std::ios_base::seekdir way, std::streamoff offset, T type)
```

Comment:
JFTR: because of the use-case, we shouldn't *need* to insert at a specific position - but a bigger fix would require a bigger refactor (and this patch fixes the appending issue).

Until then, could you please overload `Insert` for consistency?